### PR TITLE
Cleanup unused files

### LINKED ISSUE #175: Clean-up unused files
**Issue summary**

There are several files that are not included in the compilation at all, which makes it dead code.

Either include the files should be added to the compilation (be it normal operation or tests, in such case they should be run via `cargo test`) or removed.
```
[rumcajs@fedora fil-actor-states]$ rg --no-filename '^[^/].*\.rs:$' target/debug/deps/*.d | sed 's/:$//' | sort -u | diff - <(fd '\.rs$' | sort -u)
68a69
> actors/init/src/v9/types.rs
80a82
> actors/market/src/v11/testing.rs
261a264
> fil_actor_interface/build.rs
297a301,307
> fil_actors_shared/src/v10/tests/alpha_beta_filter_test.rs
> fil_actors_shared/src/v10/tests/batch_return_test.rs
> fil_actors_shared/src/v10/tests/mapmap_test.rs
> fil_actors_shared/src/v10/tests/multimap_test.rs
> fil_actors_shared/src/v10/tests/policy_test.rs
> fil_actors_shared/src/v10/tests/set_multimap_test.rs
> fil_actors_shared/src/v10/tests/set_test.rs
305a316
> fil_actors_shared/src/v11/build.rs
315a327,332
> fil_actors_shared/src/v11/tests/alpha_beta_filter_test.rs
> fil_actors_shared/src/v11/tests/batch_return_test.rs
> fil_actors_shared/src/v11/tests/mapmap_test.rs
> fil_actors_shared/src/v11/tests/multimap_test.rs
> fil_actors_shared/src/v11/tests/set_multimap_test.rs
> fil_actors_shared/src/v11/tests/set_test.rs
332a350,353
> fil_actors_shared/src/v8/tests/alpha_beta_filter_test.rs
> fil_actors_shared/src/v8/tests/multimap_test.rs
> fil_actors_shared/src/v8/tests/set_multimap_test.rs
> fil_actors_shared/src/v8/tests/set_test.rs
347a369,372
> fil_actors_shared/src/v9/tests/alpha_beta_filter_test.rs
> fil_actors_shared/src/v9/tests/multimap_test.rs
> fil_actors_shared/src/v9/tests/set_multimap_test.rs
> fil_actors_shared/src/v9/tests/set_test.rs
```
To run locally, execute `cargo clean` and then `cargo build && cargo test`; you need `rg` and `fd` installed. There are probably more solid ways of determining which files are not included in the build, but that's the first one I found. It might be worth checking it further and even introducing a lint job.

**Other information and links**

### PR DESCRIPTION
**Pull request by sudo-shashank**
**Summary of changes**
Changes introduced in this pull request:
- Cleanup unused files in all crates

**Reference issue to close (if applicable)**

Closes #175 

**Other information and links**

### CODE REVIEW

**Review by LesnyRumcajs — COMMENTED** [review]
Why did you remove the tests?

File: actors/market/src/v10/mod.rs [hunk 1]
```diff
@@ -10,7 +10,6 @@ pub use self::deal::*;
 pub use self::state::*;
 pub use self::types::*;
 
-// exports for testing
```
Comment (lemmih) [inline actors/market/src/v10/mod.rs:13]: If these exports are only used for testing, how about we delete them?

**Comment by sudo-shashank** [comment]
> Why did you remove the tests?

which ones?

**Comment by aatifsyed** [comment]
What's the methodology here?

**Comment by sudo-shashank** [comment]
> What's the methodology here?

to keep only actor state specific code

**Comment by sudo-shashank** [comment]
@lemmih what else do you think we can trim?

**Comment by sudo-shashank** [comment]
@lemmih what else do you think we can trim?

**Review by lemmih — APPROVED** [review]

**Comment by lemmih** [comment]
> @lemmih what else do you think we can trim?

I think this is fine for now.

**Comment by lemmih** [comment]
> What's the methodology here?

The `builtin-actors` crates are not supported (they're used to generate WASM code once and then abandoned). This is a problem for us since we need to access the data structures for the actors. So, we copy all the code from the builtin-actors and then we remove the smart-contract implementations while keeping the data structures. Figuring out what code can be deleted (since it'll be run via WASM) and what code should be kept isn't a completely trivial task, though.

**Comment by LesnyRumcajs** [comment]
> > Why did you remove the tests?
> 
> which ones?

Well, all of them, like `batch_return_test.rs`?

**Comment by lemmih** [comment]
> > > Why did you remove the tests?
> > 
> > 
> > which ones?
> 
> Well, all of them, like `batch_return_test.rs`.

I don't know about this test in particular but many of the tests should be removed because they're testing code that should be removed. For example, we're not using any of the smart-contract code so all smart-contract tests should be removed.

**Comment by sudo-shashank** [comment]
these are runtime util test, i guess we can keep them

**Comment by aatifsyed** [comment]
> The builtin-actors crates are not supported (they're used to generate WASM code once and then abandoned). This is a problem for us since we need to access the data structures for the actors. So, we copy all the code from the builtin-actors and then we remove the smart-contract implementations while keeping the data structures. Figuring out what code can be deleted (since it'll be run via WASM) and what code should be kept isn't a completely trivial task, though.

Thanks, this helps me understand the bigger picture here.

> Figuring out what code can be deleted (since it'll be run via WASM) and what code should be kept isn't a completely trivial task, though.

If we only need the data structures, why do we have to do any thinking at all? Just keep the code if it starts with `pub struct`?
Could we not automate the farming and deduplication of the structs?

**Review by aatifsyed — APPROVED** [review]
Going to approve this, but spin out the discussion into an issue because I think it's important

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
There are still an unused file. Please run the command I attached in the linked issue.

File: actors/market/src/v8/deal.rs [hunk 2]
```diff
@@ -149,66 +149,3 @@ pub struct DealState {
     // -1 if deal never slashed
     pub slash_epoch: ChainEpoch,
 }
-
-#[cfg(feature = "arb")]
-impl quickcheck::Arbitrary for DealProposal {
```
Comment (LesnyRumcajs) [inline actors/market/src/v8/deal.rs:154]: Why did you remove this?

File: fil_actor_interface/src/builtin/known_cids.rs [hunk 3]
```diff
@@ -111,27 +111,6 @@ mod tests {
 
     #[test]
     fn test_loading_static_value() -> Result<()> {
-        ensure!(crate::KNOWN_CIDS.actor.market.v8.contains(
```
Comment (LesnyRumcajs) [inline fil_actor_interface/src/builtin/known_cids.rs:114]: why did you remove it?

File: fil_actors_shared/src/v10/runtime/policy.rs [hunk 4]
```diff
@@ -391,55 +391,3 @@ pub mod policy_constants {
     pub const CALIBNET_MINIMUM_CONSENSUS_POWER: i64 = 32 << 30;
     pub const MAINNET_MINIMUM_CONSENSUS_POWER: i64 = 10 << 40;
 }
-
-#[cfg(feature = "arb")]
```
Comment (LesnyRumcajs) [inline fil_actors_shared/src/v10/runtime/policy.rs:395]: why did you remove it?

File: fil_actors_shared/src/v10/tests/batch_return_test.rs [hunk 5]
```diff
@@ -1,80 +1,7 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use fil_actors_shared::v10::{BatchReturn, BatchReturnGen};
```
Comment (LesnyRumcajs) [inline fil_actors_shared/src/v10/tests/batch_return_test.rs:4]: do we still need `BatchReturnGen` at all?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: fil_actors_shared/src/v10/tests/batch_return_test.rs (hunk 5 above)
Comment (sudo-shashank) [reply fil_actors_shared/src/v10/tests/batch_return_test.rs:4]: no we dont need

File: fil_actor_interface/src/builtin/known_cids.rs (hunk 3 above)
Comment (sudo-shashank) [reply fil_actor_interface/src/builtin/known_cids.rs:114]: In these we were ensuring correct values of only some versioned actor, didn't seem very useful

File: actors/market/src/v8/deal.rs (hunk 2 above)
Comment (sudo-shashank) [reply actors/market/src/v8/deal.rs:154]: this needs to be revived, my bad

File: fil_actors_shared/src/v10/runtime/policy.rs (hunk 4 above)
Comment (sudo-shashank) [reply fil_actors_shared/src/v10/runtime/policy.rs:395]: this impl is not used anywhere

**Comment by sudo-shashank** [comment]
> There are still an unused file. Please run the command I attached in the linked issue.

checked using the script, it shows only one file as of now which is `fil_actor_interface/build.rs`
but this file is executed during build process

**Comment by sudo-shashank** [comment]
@LesnyRumcajs need your review on this PR

**Review by LesnyRumcajs — APPROVED** [review]
